sentry-core/src/scope/real.rs

```diff
@@ -346,6 +348,57 @@ impl Scope {
         );
     }
 
+    /// Applies the contained scoped data to a log, setting the `trace_id` and certain default
+    /// attributes.
+    #[cfg(feature = "UNSTABLE_logs")]
+    pub fn apply_to_log(&self, log: &mut Log, send_default_pii: bool) {
+        if let Some(span) = self.span.as_ref() {
+            log.trace_id = Some(span.get_trace_context().trace_id);
+        } else {
+            log.trace_id = Some(self.propagation_context.trace_id);
+        }
+
+        if !log.attributes.contains_key("sentry.trace.parent_span_id") {
+            if let Some(span) = self.get_span() {
+                let span_id = match span {
+                    crate::TransactionOrSpan::Transaction(transaction) => {
+                        transaction.get_trace_context().span_id
+                    }
+                    crate::TransactionOrSpan::Span(span) => span.get_span_id(),
+                };
+                log.attributes.insert(
+                    "parent_span_id".to_owned(),
+                    LogAttribute(span_id.to_string().into()),
+                );
+            }
+        }
+
+        if send_default_pii {
+            if !log.attributes.contains_key("user.id") {
+                if let Some(id) = self.user().and_then(|user| user.id.as_ref()) {
```

Comment:
maybe move the `if let Some(user) = self.user()` one level up, and then check for the `contains_key` within that block.
I think that might be a bit more efficient. Though I’m unsure if `self.user()` returns a reference or a clone.